l3kernel/l3regex.dtx

```diff
@@ -1135,10 +1135,10 @@
 %   Empty a \tn{toks} or set it to a value, given its number.
 %    \begin{macrocode}
 \cs_new_protected:Npn \@@_toks_clear:N #1
-  { \@@_toks_set:Nn #1 { } }
+  { \tex_toks:D #1 = { } }
```

Comment:
How does this save time? `\@@_toks_set:Nn` _is_ `\tex_toks:D`

Reply:
one less expansion and no failed search for = . In case of `_clear`perhaps not so essential as with `_set`but ...

Reply:
Isn't the gain only from `=`, hence `\@@_toks_set:Nn #1 = { }` and `\tex_toks:D #1 = { }` are the same?
https://github.com/latex3/latex3/blob/e2f3be7d6dfe1395a80083cd68f5437cc66738dc/l3kernel/l3regex.dtx#L1136

Reply:
true as it is set equal, but I would still use :D throughout as it is more consistent and  while `\@@_toks_set:Nn #1 { }` is correct from the syntax point of view  `\@@_toks_set:Nn #1 = { }` wouldn't be.  What is the `=` doing here? picked up by 'n' or extra or ...

Reply:
Exactly @FrankMittelbach's reasoning was responsible to use `=`, which was the reason to use `:D` instead of `:Nn`.